### DISCUSSION TITLE
Please provide an example of a working before_update callback

### DISCUSSION THREAD

**Discussion by JeremyCraigMartinez** [explosion/spaCy#12656] [Help: Other Questions] [answered]
The [before_update](https://spacy.io/api/data-formats#config-training) callback was introduced in [v3.5](https://spacy.io/usage/v3-5). However, it is not clear to me how to get this to work. Every approach I've attempted has resulted in errors in the training process. How does one load a callback and pass it in as a callable? There are no working examples. All the documentation just has this set to `= null`. Since the config gets serialized, it appears to have a value of type `str`.

The following config:

```
[training]
dev_corpus = "corpora.dev"
train_corpus = "corpora.train"
...
before_to_disk = null
before_update = "module.before_update"
```

produces the error

```
Config validation error
training -> before_update   module.before_update is not callable
```

**Comment by JeremyCraigMartinez** [comment]
I also tried something with

```
import spacy

@spacy.registry.callbacks('cb_before_update')
def create_callback():
    def cb_before_update(nlp, _):
        pass
```

And referencing the callback by name, but I get the same error

**Reply by kinghuang** [reply, marked as answer]
For a registered callback function named `cb_before_update`, you can specify it like the following.

```yaml
[training]
dev_corpus = "corpora.dev"
train_corpus = "corpora.train"
...
before_to_disk = null
before_update = {"@callbacks": "cb_before_update"}
```
